javascript-practice/src/js/services/firebaseService.js

```diff
@@ -0,0 +1,55 @@
+import { initializeApp } from 'firebase/app';
+import {
+  getDatabase,
+  ref,
+  set,
+  goOffline,
+  goOnline,
+  onValue,
+} from 'firebase/database';
+import { databaseURL } from '../constants/constant';
+
+class FirebaseService {
+  constructor() {
+    const firebaseConfig = {
+      databaseURL,
+    };
+    this.app = initializeApp(firebaseConfig);
+    this.db = getDatabase(this.app);
+  }
+
+  save(data, path) {
+    return set(ref(this.db, path), data);
+  }
+
+  disconnect() {
+    goOffline(this.db);
+  }
+
+  reconnect() {
+    goOnline(this.db);
+  }
+
+  findKeyByPropery(path, property, value) {
+    return new Promise((resolve) => {
+      onValue(
+        ref(this.db, path),
+        (snapshot) => {
+          let result = false;
+          snapshot.forEach((childSnapshot) => {
```

Comment:
You mean check if any data is matching condition, we will set result = true? Should we use any? Or filter?

Reply:
What do you mean use 'any'? I have searched but there is nothing relative to 'any' in javascript. But anyway, snapshot is not an array so I can't use another method like filter, some,... It only supports forEach so I can't use any other method.